test_infra/stacks/base_stack.py

```diff
@@ -40,6 +40,27 @@ def __init__(self, scope: Construct, construct_id: str, **kwargs: str) -> None:
                 description=f"Elastic IP associated with public subnet {public_subnet.subnet_id}",
             )
 
+        # IAM Role for GitHub Open ID
```

Comment:
The `test_infra` is for external users who wish to recreate our infra to run similar tests. So I don't think creating an open id role is something they expect to be deployed. It would make more sense to have it in our private test infra instead